tests/test_material.py

```diff
@@ -0,0 +1,55 @@
+from os import path as osp
+
+import magnum as mn
+import pytest
+
+import examples.settings
+from habitat_sim.gfx import PhongMaterialInfo
+
+
+@pytest.mark.skipif(
+    not osp.exists("data/scene_datasets/habitat-test-scenes/skokloster-castle.glb")
+    or not osp.exists("data/objects/"),
+    reason="Requires the habitat-test-scenes and habitat test objects",
+)
+def test_material_api(sim):
+    cfg_settings = examples.settings.default_sim_settings.copy()
+
+    cfg_settings[
+        "scene"
+    ] = "data/scene_datasets/habitat-test-scenes/skokloster-castle.glb"
+
+    hab_cfg = examples.settings.make_cfg(cfg_settings)
+    sim.reconfigure(hab_cfg)
+    obj_mgr = sim.get_object_template_manager()
+
+    assert obj_mgr.get_num_templates() > 0
+
+    # add an object
+    object_id = sim.add_object(0)
+
+    render_asset_handle = sim.get_object_initialization_template(
+        object_id
+    ).render_asset_handle
+
+    num_materials = sim.get_num_render_asset_materials(render_asset_handle)
+    assert num_materials
+    material_index = num_materials - 1
+
+    # create a new material
+    new_material = PhongMaterialInfo()
+    new_material.shininess = 1.234
+    new_material.diffuse_color = mn.Color4(0.1, 0.2, 0.3, 0.4)
+
+    existing_material = sim.get_render_asset_material(
+        render_asset_handle, material_index
+    )
+    assert new_material != existing_material
+
+    # change one of the render asset's materials
+    sim.set_render_asset_material(render_asset_handle, material_index, new_material)
+
+    updated_material = sim.get_render_asset_material(
+        render_asset_handle, material_index
+    )
+    assert updated_material == new_material
```

Comment:
I'm reminded of how difficult it is to unit-test or integration-test graphics features. We might consider building a graphical test framework that lets us output RGB images and then automatically checks that those images don't change over time. This would be easier than trying to write asserts on pixel values.

Reply:
There is support for that kind of thing now : checkPinholeCameraRGBAObservation in SimTest.cpp.

Reply:
... and mainly the [DebugTools::CompareImage](https://doc.magnum.graphics/magnum/classMagnum_1_1DebugTools_1_1CompareImage.html) magic in Magnum :)